Update to graphite version 1.1.3

odyssey4me opened the pull request:
The latest version is 1.1.3 - the pip package list
is derived from a 'pip freeze' in a virtualenv with
the graphite-web, carbon and whisper packages
installed in it. This ensures that the packages and
everything they depend on is always the same when
deploying.

Commit:
Update to graphite version 1.1.3

The latest version is 1.1.3 - the pip package list
is derived from a 'pip freeze' in a virtualenv with
the graphite-web, carbon and whisper packages
installed in it. This ensures that the packages and
everything they depend on is always the same when
deploying.

odyssey4me commented:
It looks like the test prep script is broken. :/ I think https://github.com/nsg/ansible-role-tester/pull/7 may solve that.

odyssey4me commented:
@nsg It would appear that the failures in CI have something to do with the image setup. Do you have any ideas what's going wrong and how it may be fixed?

nsg commented:
@odyssey4me I will see what I can do. I will try to track down the problem tomorrow (in about 24 hours from now, in my timezone).

nsg commented:
FYI: I finally fixed the test images

Looks like file is missing from the debian images, you can add it in `tests/pre.yml`.

odyssey4me commented:
Great, thanks @nsg - will figure that bit out and add another commit.

Commit:
Add 'file' package for Debian

In order to be able to complete the tests successfully,
the 'file' package is required on the test system.

The package list is also changed to be ordered alphabetically.

odyssey4me commented:
OK, it looks like there are some legitimate failures going on now - I'll try and replicate them when I have some time to do so and see what needs fixing.

odyssey4me commented:
@nsg This is odd. In the test we fail for CentOS and Ubuntu Trusty, yet when I execute the test myself on a test host, I get a failure for both Debian but the others all pass. If you have a little time could you take a look and give me some pointers?

My test fails with:

```
TASK [Make sure that we got a image] ***************************************************************************************************
fatal: [images-debian-stretch]: FAILED! => {"changed": true, "cmd": "file out | grep -i image", "delta": "0:00:00.005993", "end": "2018-05-14 11:29:02.871230", "msg": "non-zero return code", "rc": 1, "start": "2018-05-14 11:29:02.865237", "stderr": "/bin/sh: 1: file: not found", "stderr_lines": ["/bin/sh: 1: file: not found"], "stdout": "", "stdout_lines": []}
fatal: [images-debian-jessie]: FAILED! => {"changed": true, "cmd": "file out | grep -i image", "delta": "0:00:00.085021", "end": "2018-05-14 11:29:02.925735", "msg": "non-zero return code", "rc": 1, "start": "2018-05-14 11:29:02.840714", "stderr": "/bin/sh: 1: file: not found", "stderr_lines": ["/bin/sh: 1: file: not found"], "stdout": "", "stdout_lines": []}
```

The travis test fails with:

```
TASK [Make sure processes are running] *****************************************
failed: [ubuntu-14-04] (item=carbon-cache) => {"changed": true, "cmd": "pgrep carbon-cache", "delta": "0:00:00.305121", "end": "2018-05-08 09:43:24.857617", "item": "carbon-cache", "msg": "non-zero return code", "rc": 1, "start": "2018-05-08 09:43:24.552496", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}
```

```
TASK [Search for errors in logs] ***********************************************
failed: [images-centos-7] (item=/var/log/nginx/error.log) => {"changed": true, "cmd": "test -f /var/log/nginx/error.log && cat /var/log/nginx/error.log", "delta": "0:00:00.027633", "end": "2018-05-08 09:43:45.561310", "failed_when_result": true, "item": "/var/log/nginx/error.log", "rc": 0, "start": "2018-05-08 09:43:45.533677", "stderr": "", "stderr_lines": [], "stdout": "2018/05/08 09:43:30 [error] 6301#0: *5 open() \"/usr/share/nginx/html/tags/tagMultiSeries\" failed (2: No such file or directory), client: 127.0.0.1, server: _, request: \"POST /tags/tagMultiSeries HTTP/1.1\", host: \"127.0.0.1\"", "stdout_lines": ["2018/05/08 09:43:30 [error] 6301#0: *5 open() \"/usr/share/nginx/html/tags/tagMultiSeries\" failed (2: No such file or directory), client: 127.0.0.1, server: _, request: \"POST /tags/tagMultiSeries HTTP/1.1\", host: \"127.0.0.1\""]}
```

nsg commented:
> My test fails with:

`file: not found`: The command file is not found.

> The travis test fails with:

The process carbon-cache is not running, and we have a few errors in the logs.

odyssey4me commented:
Abandoning this for now. I may return back to it when I have some time to figure it out.
